Lib/php/factory.i

```diff
@@ -95,7 +95,15 @@ if (!dcast) {
   Type *dobj = dynamic_cast<Type *>($1);
   if (dobj) {
     dcast = 1;
-    SWIG_SetPointerZval(return_value, SWIG_as_voidptr(dobj),$descriptor(Type *), $owner);
+    zend_object *std = NULL;
+    if ($newobj) {
+      zend_class_entry *ce = zend_lookup_class(zend_string_init("Type", sizeof("Type")-1, 0));
```

Comment:
This isn't quite right - in the example in the comment at the top, `Type` will be `Foo::Point` (and then `Foo::Circle`), and that's not going match a PHP class name.